distribution/debian/install.sh

```diff
@@ -5,62 +5,149 @@
+### Version V1.0.3 2024-01-06 - StevieTV - exit script when it is ran from install
+### Version V1.0.4 2024-04-10 - nostrusdominion - added colors, moved root check, moved architecture check, added title splash screen, improved readablity, changed app_prereq to not bother apt if they are already installed, supressed tarball extraction, add sleep timers.
 
 ### Boilerplate Warning
-#THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-#EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-#MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-#NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-#LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-#OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-#WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-scriptversion="1.0.3"
-scriptdate="2024-01-06"
+### Colors
+green='\033[0;32m'
+yellow='\033[1;33m'
+red='\033[0;31m'
+cyan='\033[0;36m'
+brown='\033[0;33m'
+reset='\033[0m' # No Color
 
-set -euo pipefail
+scriptversion="1.0.4"
+scriptdate="2024-04-10"
 
-echo "Running Sonarr Install Script - Version [$scriptversion] as of [$scriptdate]"
+set -euo pipefail
 
 # Am I root?, need root!
-
 if [ "$EUID" -ne 0 ]; then
-    echo "Please run as root."
+    echo -e ${red}"Please run as root!"
+    echo -e "Exiting script!"
     exit
 fi
 
+# Stuff the gremlins needs to know
 app="sonarr"
 app_port="8989"
 app_prereq="curl sqlite3 wget"
 app_umask="0002"
 branch="main"
 
-# Constants
-### Update these variables as required for your specific instance
+### Constants
+
+# Update these variables as required for your specific instance
 installdir="/opt"              # {Update me if needed} Install Location
 bindir="${installdir}/${app^}" # Full Path to Install Location
 datadir="/var/lib/$app/"       # {Update me if needed} AppData directory to use
 app_bin=${app^}                # Binary Name of the app
 
+# Check if architecture is correct
+echo ""
+ARCH=$(dpkg --print-architecture)
+# get arch
+dlbase="https://services.sonarr.tv/v1/download/$branch/latest?version=4&os=linux"
+case "$ARCH" in
+"amd64") DLURL="${dlbase}&arch=x64" ;;
+"armhf") DLURL="${dlbase}&arch=arm" ;;
+"arm64") DLURL="${dlbase}&arch=arm64" ;;
+*)
+    echo -e ${red}"Your arch is not supported!"
+    echo -e "Exiting installer script!"
+    exit 1
+    ;;
+esac
+
+### Title Splash!
+
+echo -e "" ${cyan}
```

Comment:
consider using a heredoc instead of multiple echos:
```shell
cat << EOF
           _____  ____  _   _          _____  _____             
          / ____|/ __ \| \ | |   /\   |  __ \|  __ \            
         | (___ | |  | |  \| |  /  \  | |__) | |__) |           
          \___ \| |  | | .   | / /\ \ |  _  /|  _  /            
          ____) | |__| | |\  |/ ____ \| | \ \| | \ \            
  _____ _|_____/_\____/|_|_\_/_/   _\_\_| _\_\_| _\_\__ _____   
...
EOF
```